### DISCUSSION TITLE
Embeddings.crop mode 'loose', 'strict' and 'center'.

### DISCUSSION THREAD

**Discussion by Gabrielk99** [pyannote/pyannote-audio#813] [Q&A] [answered]
Hello, thanks a lot for the framework

I was trying understand the crop mode of embeddings, and i was googling but, i dont found nothing about that. I was see in one issue that a type of mode like 'center' returns empty results, but i cant understand for why, and i cant understand how this modes works.

Could you explain to me, how does work and why somes modes returns empty ? Please? My thanks.

1 - How the crop mode of embeddings work
2 - the mode 'strict'
3 - the mode 'center'
4 - the mode 'loose'

**Comment by hadware** [comment, marked as answer]
Alright, i spent a significant amont of time working on `pyannote-core` (which implements most of primitives `pyannote-audio` relies on), so I might be able to answer your question. You might also want to check examples from the [documentation](http://pyannote.github.io/pyannote-core/reference.html#pyannote.core.SlidingWindow.crop). Quoting the _opus originale_:

> In ‘strict’ mode, only indices of segments fully included in ‘focus’ support are returned. In ‘loose’ mode, indices of any intersecting segments are returned. In ‘center’ mode, first and last positions are chosen to be the positions whose centers are the closest to ‘focus’ start and end times. Defaults to ‘loose’.

**Reply by Gabrielk99** [reply]
Thanks a lot, i dont know if i really understand but i'am greatfull for the your answer, and i will think about and read more examples in documentation.
